src/session.ts

```diff
@@ -357,6 +365,76 @@ export class UserSessionHandler extends BaseSessionHandler {
   }
 }
 
+export class OAuthSessionHandler extends BaseSessionHandler {
+  private readonly _oauthTokenResponse: OAuthTokenResponse
+
+  constructor (connection: MetricsConnection, oauthTokenResponse: OAuthTokenResponse) {
+    super(connection, oauthTokenResponse, false)
+
+    this._oauthTokenResponse = oauthTokenResponse
+  }
+
+  get decodedAccessToken (): any {
+    return DecodeJWT(this.accessToken) as OAuthToken
+  }
+
+  get accessToken () {
```

Comment:
You don't need to reimplement these methods as they are inherited from the `BaseSessionHandler`.

https://github.com/KeiserCorp/Keiser.Metrics.SDK/blob/b754999117a181dfdf0241bf823b4a1e342796a8/src/session.ts#L225-L260